### PR TITLE
Clean up esify and add eslint --fix

### PR DESCRIPTION
**Pull request by lemonmade**
This PR splits esify up into some more manageable chunks, and adds a final step of running `eslint --fix` on the transformed file. There are a pretty decent number of fixable linting violations that this should get rid of (most obviously: newline at the end of the file. Hurray!)

cc/ @bouk @fandy @justinthec

### CODE REVIEW

File: packages/esify/lib/eslint.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  return new Promise(function(resolve) {
+    var eslint = findESLintBinary();
+    if (eslint == null) { return; }
+    exec(eslint + ' ' + details.file + ' --fix', resolve);
+  });
+};
+
+function findESLintBinary() {
+  var possibleBinary = path.join(process.cwd(), 'node_modules/.bin/eslint');
+  return fs.statSync(possibleBinary).isFile() ? possibleBinary : null;
```
Comment (bouk) [inline packages/esify/lib/eslint.js]: you can just do `npm bin` to find the correct directory

File: packages/esify/lib/codemod.js [hunk 2]
```diff
@@ -1,6 +1,5 @@
 var path = require('path');
```
Comment (fandy) [inline packages/esify/lib/codemod.js]: The `path` and `exec` module isn't used and is causing linting errors, causing tests to fail.

File: packages/esify/lib/codemod.js [hunk 3]
```diff
@@ -1,6 +1,5 @@
...
-var decaf = require('shopify-decaf');
-var chalk = require('chalk');
+var exec = require('child_process').exec;
```
Comment (fandy) [inline packages/esify/lib/codemod.js]: Delete this line as well.

File: packages/esify/lib/eslint.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+var exec = require('child_process').exec;
+var path = require('path');
+var fs = require('fs');
+
+module.exports = function runESLint(details) {
+  return new Promise(function(resolve) {
```
Comment (fandy) [inline packages/esify/lib/eslint.js]: I think we need the `es2015` preset on `babel-preset-shopify` in order to define `Promise`.

Comment (lemonmade) [inline packages/esify/lib/eslint.js]: It was actually a linting environment thing that needed to be fixed. Thanks for catching that.

**Pull request merged**